pipelines.md

```diff
@@ -0,0 +1,323 @@
+
+   ```shell
+   cd ./pipelines/sample-helper-files/./manual-pipeline-run-script.sh -r [git_repo of the Appsody project] -i [docker registery path of the image to be created] -c [application stack name of which pipeline to be run]"
+   ```
+
+   - The following example is configured to use the dockerhub container registry:
+
+      ```shell
+       ./manual-pipeline-run-script.sh -r https://github.com/mygitid/appsody-test-project -i index.docker.io/mydockeid/my-java-microprofile-image -c java-microprofile"
+      ```
+
+   - The following example is configured to use the local OpenShift container registry:
+
+      ```shell
+       ./manual-pipeline-run-script.sh -r https://github.com/mygitid/appsody-test-project -i docker-registry.default.svc:5000/kabanero/my-java-microprofile-image -c java-microprofile"
+      ```
+
+### Running pipelines manually from the command line
+
+1. Login to your cluster. For example,
+
+   ```shell
+   oc login <master node IP>:8443
+   ```
+
+1. Clone the pipelines repo.
+
+   ```shell
+   git clone https://github.com/kabanero-io/kabanero-pipelines
+   cd kabanero-pipelines
+   ```
+
+1. Create pipeline resources.
+
+   Use the `pipeline-resource-template.yaml` file to create the `PipelineResources`. The `pipeline-resource-template.yaml` is provided in the pipelines [/pipelines/sample-helper-files](https://github.com/kabanero-io/kabanero-pipelines/tree/master/pipelines/sample-helper-files) directory. Update the docker-image URL. You can use the sample GitHub repo or update it to point to your own GitHub repo.
+
+1. After you update the file, apply it as shown in the following example:
+
+   ```shell
+   oc apply -f <stack-name>-pipeline-resources.yaml
+   ```
+
+### Activating tasks and pipelines
+
+The installations that activate the featured application stacks also activate the tasks and pipelines. If you are creating a new task or pipeline, activate it manually, as shown in the following example.
+
+```shell
+oc apply -f <task.yaml>
+oc apply -f <pipeline.yaml>
+```
+
+### Running the pipeline
+
+A sample `manual-pipeline-run-template.yaml` file is provided in the [/pipelines/sample-helper-files](https://github.com/kabanero-io/kabanero-pipelines/tree/master/pipelines/sample-helper-files) directory. Rename the template file to a name of your choice (for example, pipeline-run.yaml), and update the file to replace `application-stack-name` with the name of your application stack. After you update the file, run it as shown in the following example.
+
+```shell
+oc apply -f <application-stack-name>-pipeline-run.yaml
+```
+
+<!--
+// =================================================================================================
+// Running pipelines from the command line for your custom built application stacks
+// =================================================================================================
+-->
+
+## Running pipelines from the command line for your custom built application stacks
```

Comment:
This whole section has the same problem with headers as described previously. However, reading and understanding the steps I think the section can be replaced with a simpler version that does not require so many sections:

**FROM HERE:**

The following steps explain how to run pipelines against custom built application stack images instead of the provided application stacks.

## Running pipelines from the command line for your custom built application stacks

The following steps explain how to run pipelines against custom built application stack images instead of the provided application stacks.

### Setting up a container registry URL for the custom application stack image

By default, pipelines pull the application stack images from Docker hub. If you are publishing your application stack images to any other registry, use the following process to configure the custom repository from which your pipelines pull the container images.

1. After you clone the `kabanero-pipelines` repository, find the `stack-image-registry-map.yaml` configmap template file. Add your container registry URL to this file in place of the `default-stack-image-registry-url` statement.

    ```
    cd kabanero-pipelines/pipelines/common/
    vi stack-image-registry-map.yaml
    ```

2. If your custom application stack image is stored in an internal OpenShift registry, the service account that is associated with the pipelines must be configured to allow the pipelines to pull from the internal registry without configuring a secret. If your custom application stack is stored in a container registry with an external route, follow these steps to set up a Kubernetes secret:

    - Find the `default-stack-image-registry-secret.yaml` template file in the cloned kabanero-pipelines repo (`kabanero-pipelines/pipelines/common`) and update it with the username and token password for the container registry URL you specified previously.

    - Create a Base64 format version of the username and password for the external route container registry URL.

    ```
     echo -n <your-registry-username> | base64
     echo -n <your-registry-password> | base64
    ```

    - Update the `default-stack-image-registry-secret.yaml` file with the Base64 formatted username and password.

    ```
     vi default-stack-image-registry-secret.yaml
    ```   

    - Apply the `default-stack-image-registry-secret.yaml` file to the cluster

    ```
     oc apply -f default-stack-image-registry-secret.yaml
    ``` 

3. Apply the following configmap file, which will set your container registry.

    ```
    oc apply -f stack-image-registry-map.yaml
    ```

Reply:
We should get @kvijai82 to confirm this is ok.

Reply:
I've reordered the steps as you suggest.

Reply:
Could we please update the two `kabanero-pipelines/pipelines/common/` references to `kabanero-pipelines/pipelines/sample-helper-files/`

Reply:
Also in the section that has this, we would need to update it to reflect a change we just merged an hour ago:


```By default, pipelines pull the application stack images from Docker hub. If you are publishing your application stack images to any other registry, use the following process to configure the custom repository from which your pipelines pull the container images.```

to

```By default, pipelines pull the application stack images from Docker hub. If you are publishing your application stack images to any other registry there are two ways to tell the pipeline where to look for it.  

- You can specify it in your project in the `.appsody-config.yaml` file.  When you initialize the stack appsody (in CLI versions 0.5.8 and above) by default will update the stack in the `.appsody-config.yaml` file to also include the registry.  You can pass in the registry info using the `--stack-registry` option in the `appsody init` command.

- You could also create a config map as described below to specify the custom registry from which your pipelines pull the container images.  

NOTE: If a value is specified in both the config map and in the .appsody-config.yaml and they are different, the config map will take precedence. ```

Reply:
@kvijai82 Can you confirm the latest change is ok?